feat: Add README file (#28)
kryoniteorg/kryo-discord-whitelist: README.md

<div align="center">
  <a href="https://github.com/kryoniteorg/kryo-discord-whitelist">
    <img src="https://raw.githubusercontent.com/kryoniteorg/.github/main/assets/kryonite_logo.svg" alt="Kryonite logo" width="80" height="80">
  </a>
</div>

<h1 align="center">kryo-discord-bot</h1>
<div align="center">

>     A Discord bot to whitelist players on a <a href="https://github.com/PaperMC/Velocity">Velocity</a> with multi-proxy support.

Reviewer: ```suggestion
    A Discord bot to whitelist players on <a href="https://github.com/PaperMC/Velocity">Velocity</a> with multi-proxy support.
```

    <img alt="Maintainability Rating" src="https://sonarcloud.io/api/project_badges/measure?project=kryoniteorg_kryo-discord-whitelist&metric=sqale_rating">
    <img alt="Reliability Rating" src="https://sonarcloud.io/api/project_badges/measure?project=kryoniteorg_kryo-discord-whitelist&metric=reliability_rating">
    <img alt="Security Rating" src="https://sonarcloud.io/api/project_badges/measure?project=kryoniteorg_kryo-discord-whitelist&metric=security_rating">
    <br />
    <br />
</div>

## About the project

kryo-discord-whitelist is a whitelist where Discord users can add themselves by writing their username into a specific channel.

## Prerequisites
kryo-discord-whitelist needs the following services to be installed and configured to function properly:

- [MariaDB](https://mariadb.org/) database
- [RabbitMQ](https://www.rabbitmq.com/) message broker

It is not recommended using the root user of the [MariaDB](https://mariadb.org/) server for kryo-discord-whitelist. Please create an extra database with an extra user that is limited to that database.

## Setup

> The installation and usage of kryo-discord-whitelist needs to separate modules, the Discord bot itself and a [Velocity](https://github.com/PaperMC/Velocity) plugin. At the moment you need to build all JAR files yourself because we do not distribute prebuilt JARs.  

Reviewer: ```suggestion
The installation and usage of kryo-discord-whitelist needs two separate modules, the Discord bot itself and a [Velocity](https://github.com/PaperMC/Velocity) plugin. At the moment you need to build all JAR files yourself because we do not distribute prebuilt JARs.  
```

1. Goto the [Discord Developer Portal](https://discord.com/developers/applications) and create a new Application.
2. Click on "Bot" in the sidemenu and add a new bot.
3. Click "Copy" next to the profile picture of the bot to copy the token.

With the generated token we can now start the discord bot. The discord bot needs two environment variables or Java start parameters to start.

| Environment variable | Start parameter     | Description                                              |
|----------------------|---------------------|----------------------------------------------------------|
| TOKEN                | -DTOKEN             | Discord Bot token                                        |
| CONNECTION_STRING    | -DCONNECTION_STRING | Connection String for connecting to the MariaDB database |

A startup command could look like the following:
```bash
java -Xms128M -Xmx512M -DTOKEN=DISCORD_TOKEN -DCONNECTION_STRING=jdbc:mariadb://localhost:3306/database?user=user&password=password -JAR kryo-discord-whitelist-bot.JAR
```

### Velocity plugin
To install the [Velocity](https://github.com/PaperMC/Velocity) plugin just copy the JAR-file into the plugin directory.

> Furthermore, the [Velocity](https://github.com/PaperMC/Velocity) plugin needs some environment variables or start parameters too.

Reviewer: ```suggestion
Furthermore, the [Velocity](https://github.com/PaperMC/Velocity) plugin needs some environment variables. Those can also be provided as startup parameters if the usage of environment variables is not possible.
```
